cmake/AwsGetCmdArguments.cmake

```diff
@@ -0,0 +1,49 @@
+# Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+# SPDX-License-Identifier: Apache-2.0.
+
+# Get list of variables provided via command line arguments (i.e. -DVARIABLE=VALUE). This function iterates over all
+# variables and determines if they're provided via command line or not by checking their help strings. CMake sets help
+# strings for variables provided via command line to the "No help, variable specified on the command line." phrase since
+# at least v3.0.
+# Via https://cmake.org/pipermail/cmake/2018-January/067002.html
+# Arguments:
+#  VARS_TO_IGNORE Variables that should be ignored even if they were provided via command line. Multiple variables can
+#                 be specified separated by space.
+#
+# MUST be done before call to 'project'. The reason is that project() resets help strings for some variables
```

Comment:
How would the requirement to call it before `project()` work with aws-sdk-cpp, which builds CRT via [add_subdirectory(crt/aws-crt-cpp)
](https://github.com/aws/aws-sdk-cpp/blob/4b5c0cf73ef3929858bfe7460bc72983a1dc98a3/CMakeLists.txt#L231)

Reply:
The most important variables are set explicitly in `aws_prebuild_dependency`. This PR adds `CMAKE_TOOLCHAIN_FILE`, which should handle the simplest cross-compiling use cases even without forwarding command-line arguments.

If we had to deal only with variables listed here (https://cmake.org/cmake/help/latest/manual/cmake-variables.7.html), we could probably just check them all inside `aws_prebuild_dependency`. But various platforms can add custom variables.
For example, `gradle` in the Android CI job in this repo passes multiple `ANDROID_*` variables (described [here](https://developer.android.com/ndk/guides/cmake#variables)) via command line (that's where this "gather cmd vars" idea came from).

Reply:
I removed parsing cmd vars